mpld3/js/mpld3.v0.3git.js

```diff
@@ -580,9 +581,24 @@
     delete pathProps.color;
     pathProps.edgewidth = pathProps.linewidth;
     delete pathProps.linewidth;
+    drawstyle = pathProps.drawstyle;
+    delete pathProps.drawstyle;
     this.defaultProps = mpld3_Path.prototype.defaultProps;
     mpld3_Path.call(this, ax, pathProps);
-    this.datafunc = d3.svg.line().interpolate("linear");
+    alert(drawstyle);
+    switch (drawstyle) {
+     case "steps":
+     case "steps-pre":
+      this.datafunc = d3.svg.line().interpolate("step-before");
+      break;
+
+     case "steps-post":
+      this.datafunc = d3.svg.line().interpolate("step-after");
+      break;
+
```

Comment:
There have been some requests for `steps-mid` support, and I bet this is just where it would be added.  I know it's harder than `steps-pre` and `steps-post`, so no need to add it now... I'm just trying to plan ahead, while this is fresh in your mind.